pkg/services/dns/hosts_file.go

```diff
@@ -0,0 +1,94 @@
+package dns
+
+import (
+	"net"
+	"sync"
+
+	"github.com/areYouLazy/libhosty"
+	"github.com/fsnotify/fsnotify"
+	log "github.com/sirupsen/logrus"
+)
+
+type HostsFile interface {
+	LookupByHostname(name string) (net.IP, error)
+}
+
+type hosts struct {
+	hostsReadLock sync.RWMutex
+	hostsFilePath string
+	hostsFile     *libhosty.HostsFile
+}
+
+// NewHostsFile Creates new HostsFile instance
+// Pass ""(empty string) if you want to use default hosts file
+func NewHostsFile(hostsPath string) (HostsFile, error) {
+	hostsFile, err := readHostsFile(hostsPath)
+	if err != nil {
+		return nil, err
+	}
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+
+	h := &hosts{
+		hostsFile:     hostsFile,
+		hostsFilePath: hostsFile.Config.FilePath,
+	}
+	go func() {
+		h.startWatch(watcher)
+	}()
+	return h, nil
+}
+
+func (h *hosts) startWatch(w *fsnotify.Watcher) {
+	err := w.Add(h.hostsFilePath)
```

Comment:
Will we be affected by the issue described at https://github.com/fsnotify/fsnotify?tab=readme-ov-file#watching-a-file-doesnt-work-well when `crc-admin-helper` rewrites the `hosts` file?

Reply:
Did you check that it's fine? The code hasn't changed, and I don't think you've answered this comment?

Reply:
It is addressed, looks like GH not refreshed that change.
Look on https://github.com/containers/gvisor-tap-vsock/pull/401/files#diff-a347cf19358b6ed56129a699632299bef3dd2899b0e69404843f4e66839f44b7R42

Reply:
Ah I see the changes on github now as well, thanks!